bitwarden_license/test/Commercial.Core.Test/SecretsManager/AccessPolicies/DeleteAccessPolicyCommandTests.cs

```diff
@@ -11,28 +15,241 @@
+                policyToReturn =
+                    new GroupProjectAccessPolicy { Id = data, GrantedProjectId = grantedProject.Id, Group = mockGroup, GrantedProject = grantedProject };
+                break;
+            case AccessPolicyType.ServiceAccountProjectAccessPolicy:
+                policyToReturn = new ServiceAccountProjectAccessPolicy
+                {
+                    Id = data,
+                    GrantedProjectId = grantedProject.Id,
+                    ServiceAccount = mockServiceAccount,
+                    GrantedProject = grantedProject,
+                };
+                break;
+        }
+
+        sutProvider.GetDependency<IAccessPolicyRepository>().GetByIdAsync(data)
+            .Returns(policyToReturn);
+
+        await Assert.ThrowsAsync<NotFoundException>(() =>
+            sutProvider.Sut.DeleteAsync(data, userId));
+
+        await sutProvider.GetDependency<IAccessPolicyRepository>().DidNotReceiveWithAnyArgs().DeleteAsync(default);
+    }
+
+    [Theory]
+    [BitAutoData(AccessPolicyType.UserServiceAccountAccessPolicy, PermissionType.RunAsAdmin)]
+    [BitAutoData(AccessPolicyType.UserServiceAccountAccessPolicy, PermissionType.RunAsUserWithPermission)]
+    [BitAutoData(AccessPolicyType.GroupServiceAccountAccessPolicy, PermissionType.RunAsAdmin)]
+    [BitAutoData(AccessPolicyType.GroupServiceAccountAccessPolicy, PermissionType.RunAsUserWithPermission)]
+    public async Task DeleteAccessPolicy_ServiceAccountGrants_PermissionsCheck_Success(
+        AccessPolicyType accessPolicyType,
+        PermissionType permissionType,
+        Guid data,
+        Guid userId,
+        ServiceAccount grantedServiceAccount,
+        Group mockGroup,
+        SutProvider<DeleteAccessPolicyCommand> sutProvider)
+    {
+        sutProvider.GetDependency<ICurrentContext>().AccessSecretsManager(Arg.Any<Guid>()).Returns(true);
+        BaseAccessPolicy policyToReturn = null;
+        switch (accessPolicyType)
+        {
+            case AccessPolicyType.UserServiceAccountAccessPolicy:
+                policyToReturn =
+                    new UserServiceAccountAccessPolicy
+                    {
+                        Id = data,
+                        GrantedServiceAccountId = grantedServiceAccount.Id,
+                        GrantedServiceAccount = grantedServiceAccount,
+                    };
+                break;
+            case AccessPolicyType.GroupServiceAccountAccessPolicy:
+                mockGroup.OrganizationId = grantedServiceAccount.OrganizationId;
+                policyToReturn =
+                    new GroupServiceAccountAccessPolicy
+                    {
+                        Id = data,
+                        GrantedServiceAccountId = grantedServiceAccount.Id,
+                        Group = mockGroup,
+                        GrantedServiceAccount = grantedServiceAccount,
+                    };
+                break;
+        }
```

Comment:
Extract?

Reply:
Resolved in https://github.com/bitwarden/server/pull/2628/commits/79bce819b50d75b5ce021b4b4dfad21dc9c97cd8